catalog/plugin/catalog-plugin-expirationdate/src/main/java/org/codice/ddf/catalog/plugin/expiration/ExpirationDatePlugin.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.expiration;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+/**
+ * Pre-Ingest Plugin to overwrite Metacard expiration dates.
+ */
+public class ExpirationDatePlugin implements PreIngestPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExpirationDatePlugin.class);
+
+    private int offsetFromModifiedDate;
+
+    private boolean overwriteIfBlank;
+
+    private boolean overwriteIfExists;
+
+    /**
+     * Modify metacard expiration dates (pre-ingest).
+     */
+    @Override
+    public CreateRequest process(CreateRequest createRequest)
+        throws PluginExecutionException, StopProcessingException {
+
+        LOGGER.debug("START Pre-Ingest Plugin: {}.process(CreateRequest createRequest)",
+                this.getClass().getName());
+
+        if (createRequest == null || createRequest.getMetacards() == null
+                || createRequest.getMetacards().isEmpty()) {
+            throw new PluginExecutionException("No metacards to validate or CreateRequest is null");
+        }
+
+        List<Metacard> metacards = createRequest.getMetacards();
+
+        for (Metacard metacard : metacards) {
+            updateExpirationDate(metacard);
+        }
+
+        LOGGER.debug("END Pre-Ingest Plugin: {}.process(CreateRequest createRequest)",
+                this.getClass().getName());
+        return createRequest;
+    }
+
+    @Override
+    public UpdateRequest process(UpdateRequest updateRequest)
```

Comment:
Question for everyone - should an update extend the expiration date?  @millerw8 ?